### DISCUSSION TITLE
Parsing LF

### DISCUSSION THREAD

**Discussion by schoeberl** [lf-lang/lingua-franca#1454] [General]
Related to the discussion today on dependency on frameworks:

When working on a language, we are so used to using parser generators, starting with lex/yacc, later ANTLR, or now Xtext. We forget that parsing a language is not rocket science. We could write a parser for LF directly in Java, avoiding all generation of parser code. This would reduce our dependency on external tools and avoid the generation of source files for our compiler. I recently read a very nice book on this: https://craftinginterpreters.com/, which teaches exactly this. I highly recommend this book ;-)

**Comment by petervdonovan** [comment]
The awkwardness in using newlines as expression terminators in our XText grammar is one example of this and has been the subject of some discussion in the syntax channel on Zulip. I am still in favor of this idea, in principle, but unfortunately we have a consensus that this is not on the horizon right now. Because we use the Ecore model everywhere in the compiler, and because we use XText features in our IDE support, this will be a highly disruptive change.
